FLT/DivisionAlgebra/Finiteness.lean

```diff
@@ -164,8 +166,25 @@ lemma C_compact : IsCompact (C K D) := by
 lemma antidiag_mem_C {β : D_𝔸ˣ} (hβ : β ∈ ringHaarChar_ker D_𝔸) :
     ∃ b ∈ Set.range (incl K D : Dˣ → D_𝔸ˣ),
     ∃ ν ∈ ringHaarChar_ker D_𝔸,
-    β = b * ν ∧ ((ν : D_𝔸), ((ν⁻¹ : D_𝔸ˣ) : D_𝔸)) ∈ C K D :=
-  sorry
+    β = b * ν ∧ ((ν : D_𝔸), ((ν⁻¹ : D_𝔸ˣ) : D_𝔸)) ∈ C K D := by
+  obtain ⟨x1, hx1, b1, ⟨b1, rfl⟩, eq1⟩ := X_meets_kernel K D hβ
+  obtain ⟨x2, hx2, b2, ⟨b2, rfl⟩, eq2⟩ := X_meets_kernel' K D hβ
+  obtain ⟨x1, rfl⟩ : IsUnit x1 := ⟨↑β⁻¹ * incl K D b1,
+    Eq.symm ((Units.eq_inv_mul_iff_mul_eq β).mpr eq1)⟩
```

Comment:
You don't *have* to golf everything to oblivion in FLT :-) That's just the mathlib culture. But given that you mentioned it, I'm assuming that the type of `((Units.eq_inv_mul_iff_mul_eq β).mpr eq1)` is `X = Y` i.e. `Eq X Y` which means that you can golf this line to ` ((Units.eq_inv_mul_iff_mul_eq β).mpr eq1).symm⟩`.